restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -589,6 +595,65 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
 
     }
 
+    /**
+     * FIXME: duplicated code make into static function or something more optimized
+     * Replace headers
+     * @param SipServletRequest message
+     * @param Map<String, ArrayList<String> > headers
+     */
+    private void addHeadersToMessage(SipServletRequest message, Map<String, ArrayList<String> > headers) {
```

Comment:
Also @abdulazizali77, we definitely need to surround this in a try catch clause in order to catch exceptions when user will try to pass/modify system headers. In that case, Sip Servlet will not allow the modification and will throw an exception.